sql/src/parser.rs

```diff
@@ -593,6 +579,78 @@ impl<'a> Parser<'a> {
         }))
     }
 
+    fn maybe_parse_and_check_key_partition(
+        &mut self,
+        columns: &[ColumnDef],
+    ) -> Result<Option<KeyPartition>> {
+        let linear = if self.consume_token("KEY") {
+            false
+        } else if self.consume_tokens(&["LINEAR", "KEY"]) {
+            true
+        } else {
+            return Ok(None);
+        };
+
+        let key_columns = self.parser.parse_parenthesized_column_list(Mandatory)?;
+
+        // Ensure at least one column for partition key.
+        if key_columns.is_empty() {
+            return parser_err!(
+                "except at least one partition key, default partition key is unsupported now"
+                    .to_string()
+            );
+        }
+
+        // Validate all columns composing partition key:
+        //  - The column must exist;
+        //  - The column must be a tag;
+        for key_col in &key_columns {
+            let col_def = match columns.iter().find(|c| c.name.value == key_col.value) {
+                Some(v) => v,
+                None => {
+                    return parser_err!(format!(
+                        "partition key contains non-existent column:{}",
+                        key_col.value,
+                    ))
+                }
+            };
+            let tag_option = col_def.options.iter().find(|opt| {
+                opt.option == ColumnOption::DialectSpecific(vec![Token::make_keyword(TAG)])
+            });
+            if tag_option.is_none() {
+                return parser_err!(format!(
+                    "partition key must be tag, key name:{:?}",
+                    key_col.value
+                ));
+            }
+        }
```

Comment:
The check logic here is same as hash partition, maybe extract a common function is better?